### PR TITLE
Clean up tech debt.

### PR DESCRIPTION
**Pull request by efoxepstein**
- Improves unit tests to test the class under test instead of
dependencies.

- Simplify Kotlin.

---
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31)

### CODE REVIEW

**Review by stevenwarejones — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31)* status: 0 of 27 files reviewed, 5 unresolved discussions (waiting on @efoxepstein)

___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 42 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3lNPyAGIeZkyPYxjC:-Md3lNPyAGIeZkyPYxjD:b-8g4mdx) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L42)):*
> ```Kotlin
>   }
> 
>   /** Reads a single blob from either [sharedStorage] or [privateStorage] as specified in [step]. */
> ```

so either it should require only one be present or it should check both
___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 63 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3kySe4x7UNtgek4iu:-Md3kySe4x7UNtgek4iv:b9ce2m0) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L63)):*
> ```Kotlin
> 
>   override suspend fun execute(input: Map<String, ByteString>): Map<String, ByteString> {
>     require(input.isEmpty())
> ```

this implies the exchange workflow was invalid. Shouldn't this be in validate step?
___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 66 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3liG79CEO-ngmUt47:-Md3liG87JUk-DL0ZMjo:b-jhcfpm) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L66)):*
> ```Kotlin
> 
>     while (true) {
>       if (throttler.onReady { isReady() }) {
> ```

feels a little weird to have a throttler inside an `while (true)` don't you think? It might be better to write a custom throttler that does both things together.
___
*[src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutor.kt, line 0 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3pHoD3LEpCLoaxeXH:-Md3pHoD3LEpCLoaxeXI:b5u2347) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutor.kt#L0)):*
was git not able to track the rename here?
___
*[src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutorImpl.kt, line 47 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3p6-U526A3nGmFx1H:-Md3p6-V0BVjWGyQpCF3:budqesp) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutorImpl.kt#L47)):*
> ```Kotlin
>   /** Reads inputs for [step], executes [step], and writes the outputs to appropriate [Storage]. */
>   override suspend fun execute(attemptKey: ExchangeStepAttemptKey, step: ExchangeWorkflow.Step) {
>     withContext(CoroutineName(attemptKey.exchangeStepAttemptId)) {
> ```

I was thinking `withContext` should wrap getting the input and writing the output as well

**Review by efoxepstein — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31)* status: 0 of 27 files reviewed, 5 unresolved discussions (waiting on @efoxepstein and @stevenwarejones)

___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 42 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3lNPyAGIeZkyPYxjC:-Md3sKw23UAtS9TO1clZ:b-c1f5z2) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L42)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

so either it should require only one be present or it should check both
</blockquote></details>

In the `init` block, it ensures this is true.
___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 66 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3liG79CEO-ngmUt47:-Md3sTGH8mSDPgqH2cIo:b-1r32zt) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L66)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

feels a little weird to have a throttler inside an `while (true)` don't you think? It might be better to write a custom throttler that does both things together.
</blockquote></details>

Isn't this how throttlers are usually used? 
Is the meaning of this code unclear or hard to follow?
___
*[src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutor.kt, line  at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3pHoD3LEpCLoaxeXH:-Md3tDTa-edC347aPwMD:bwevvus) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutor.kt#L)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

was git not able to track the rename here?
</blockquote></details>

I guess?
___
*[src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutorImpl.kt, line 47 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3p6-U526A3nGmFx1H:-Md3tFO282dR6xeIj2BQ:b3qtlrt) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/launcher/ExchangeTaskExecutorImpl.kt#L47)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

I was thinking `withContext` should wrap getting the input and writing the output as well
</blockquote></details>

It does, doesn't it?

**Review by efoxepstein — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31)* status: 0 of 27 files reviewed, 5 unresolved discussions (waiting on @stevenwarejones)

___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 63 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3kySe4x7UNtgek4iu:-Md4ACtl8dPnAclKgdht:bss4a2b) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L63)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

this implies the exchange workflow was invalid. Shouldn't this be in validate step?
</blockquote></details>

Removed.

**Review by stevenwarejones — APPROVED** [review]
> Reviewed 3 of 27 files at r1.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31)* status: 3 of 27 files reviewed, 5 unresolved discussions (waiting on @efoxepstein and @stevenwarejones)

___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 66 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3liG79CEO-ngmUt47:-MdINfAs4Le7kndCjL4D:b-pv2qmm) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L66)):*
<details><summary><i>Previously, efoxepstein (Eli Fox-Epstein) wrote…</i></summary><blockquote>

Isn't this how throttlers are usually used? 
Is the meaning of this code unclear or hard to follow?
</blockquote></details>

i need to think about this more but good enough for now

**Review by efoxepstein — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31)* status: 3 of 27 files reviewed, 5 unresolved discussions (waiting on @stevenwarejones)

___
*[src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt, line 66 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/panel-exchange-client/31#-Md3liG79CEO-ngmUt47:-MdIOdSJ4xghnL_CcgZA:b-ucfknh) ([raw file](https://github.com/world-federation-of-advertisers/panel-exchange-client/blob/7096eb856ece9d672d0919947a940b90df331438/src/main/kotlin/org/wfanet/panelmatch/client/exchangetasks/InputTask.kt#L66)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

i need to think about this more but good enough for now
</blockquote></details>

Thanks!
